packages/frameworks/src/types.ts

```diff
@@ -1,50 +1,174 @@
+  /**
+   * A marketing tagline for the framework
+   * @example "Next.js makes you productive with React instantly — whether you want to build static or dynamic sites."
+   */
   tagline?: string;
+  /**
+   * A URL to the official website of the framework
+   * @example https://nextjs.org
+   */
   website?: string;
+  /**
+   * Short description of the framework
+   * @example A Next.js app and a Serverless Function API.
+   */
   description: string;
+  /**
+   * A ponderated value to sort matching frameworks
+   * @example 1
+   */
   sort?: number;
-  useRuntime?: { src: string; use: string };
+  /**
+   * Runtime configuration required to run the framework in Vercel
+   */
+  useRuntime?: {
+    /**
+     * Runtime source
+     * @example "package.json"
+     */
+    src: string;
+    /**
+     * Runtime
+     * @example @vercel/next
+     */
+    use: string;
+  };
   ignoreRuntimes?: string[];
+  /**
+   * Detectors used to find out the framework
+   */
   detectors?: {
+    /**
+     * Collection of detectors that must be matched for the framework
+     * to be detected.
+     */
     every?: FrameworkDetectionItem[];
+    /**
+     * Collection of detectors where one match triggers the framework
+     * to be detected.
+     */
     some?: FrameworkDetectionItem[];
   };
   settings: {
+    /**
+     * Default install command or a placeholder
+     */
     installCommand: Setting;
+    /**
+     * Default build command or a placeholder
+     */
     buildCommand: Setting;
+    /**
+     * Default dev command or a placeholder
+     */
     devCommand: Setting;
+    /**
+     * Default Output Directory
+     */
     outputDirectory: Setting;
   };
+  /**
+   * A list of recommended integrations for the framework
+   */
   recommendedIntegrations?: {
+    /**
+     * Id of the recommended integration
+     * @example oac_5lUsiANun1DEzgLg0NZx5Es3
+     */
     id: string;
+    /**
+     * Dependencies of the recommended integration
+     * @example ["next-plugin-sentry", "next-sentry-source-maps"]
+     */
     dependencies: string[];
   }[];
-
+  /**
+   * Name of a dependency in `package.json` to detect this framework.
+   * @example hexo
+   */
   dependency?: string;
+  /**
+   * Function that returns the name of the directory that the framework outputs
+   * its build results to. In some cases this is read from a configuration file.
+   * @example async () => 'public'
+   */
   getOutputDirName: (dirPrefix: string) => Promise<string>;
+  /**
+   * An array (or a function that returns an array) of default `Route` rules that
+   * the framework uses.
+   * @example [{ handle: 'filesystem' }, { src: '.*', status: 404, dest: '404.html' }]
+   */
   defaultRoutes?: Route[] | ((dirPrefix: string) => Promise<Route[]>);
+  /**
+   * A glob string of files to cache for future deployments.
+   * @example .nuxt/**
+   */
   cachePattern?: string;
+  /**
+   * The default build command for the framework.
+   * @example nuxt generate
```

Comment:
```suggestion
   * @example next build
```

Let's stick with Next.js for the example to be consistent